src/browser/uuid_availability.ts

```diff
@@ -0,0 +1,22 @@
+const electron = require('electron');
+const namedMutex = electron.namedMutex;
+import { meshEnabled } from './connection_manager';
+import { getAppRunningState } from './core_state';
+
+const makeMutexKey = (uuid: string) => `uuid-${uuid}`;
+
+export function isUuidAvailable(uuid: string) {
+    return !getAppRunningState(uuid) && (
+        !meshEnabled || (
+            namedMutex.tryLock(makeMutexKey(uuid)) === 0
+        )
+    );
+}
+
+export function releaseUuid (uuid: string) {
+    const key = makeMutexKey(uuid);
+    let released = namedMutex.releaseLock(key);
+    while (!released) {
```

Comment:
Apologies for not looking into the namedMutex api ahead of time, but can you say a few nice words regarding `namedMutex.releaseLock` and how we are guaranteed to have it be released before too long? Just want to make sure that `while (!released)` can't get into a forever loop.

Reply:
@anguslim

Reply:
So in the case where we successfully release the lock (no errors encountered) we will try to release it one more time and let the error code or -1 break the loop? Apologies if I'm missing something.
```
    let released = namedMutex.releaseLock(key); // = 0 == false
    while (!false) {
        released = namedMutex.releaseLock(key); // = some error code that breaks the loop?
    }


```

Reply:
Assuming no errors are encountered, the while loop will run until a release fails. As soon as the lock is completely released (i.e. the lock was released X times after being acquired X times), the next release attempt will result in an error, either within the Windows API (so some Windows error code is returned) or because the mutex has been cleaned up and no longer exists (so -1 is returned).

Reply:
ahh gotcha, thanks!

Reply:
this seems like a rather un-standard way to use a mutex, maybe even has a bit of a code smell to it. is this the only place in code that the `releaseLock` is called? if not, doing so can cause breaks elsewhere.

the standard pattern is to call release once per acquire. im also curious as to why we would be attempting to acquire a mutex that is already held (i.e. multiple times). this is typically only done when passing through a synchronous callstack and each acquire ends with a release.

Reply:
The way the code is written the mutex will only be acquired once. The while is there as a backup. We did encounter a case where even though the release call returned ok, the mutex hadn't been deleted hence keeping the while.